snapshot/snapshots/fake_filer.go

```diff
@@ -88,3 +92,67 @@ func (c *initingCheckouter) Checkout(id string) (snapshot.Checkout, error) {
 		id:   id,
 	}, nil
 }
+
+// Creates a new Ingester that will copy ingested paths to its own tmp snapshots dir.
+func MakeTempIngester(tmp *temp.TempDir) snapshot.Ingester {
```

Comment:
Your changes to this file seem like they're new functionality; think it's worth putting in a new file?